sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/component.py

```diff
@@ -223,11 +224,22 @@ def _get_snapshot_id(
         return snapshot_id
 
     @contextmanager
-    def _resolve_local_code(self):
-        """Create a Code object pointing to local code and yield it."""
-        # Note that if self.code is already a Code object, this function won't be called
-        # in create_or_update => _try_resolve_code_for_component, which is also
-        # forbidden by schema CodeFields for now.
+    def _resolve_local_code(self) -> Optional[Code]:
+        """Try to create a Code object pointing to local code and yield it.
+        If there is no local code to upload, yield None.
+        Otherwise, yield a Code object pointing to the code.
+        """
+        # an internal component always has a default local code of its base path
+        # otherwise, if there is no local code, yield super()._resolve_local_code() and return early
+        if self.code is not None:
+            with super()._resolve_local_code() as code:
+                if not isinstance(code, Code) or code._is_remote:
+                    yield code
+                    return
+
+        # This is forbidden by schema CodeFields for now so won't happen.
+        if isinstance(self.code, Code):
+            raise ValueError("Code is a private class and can't be set to component.code directly.")
```

Comment:
Why do we need this exception here? Will there be cases when `_resolve_local_code` called on component with code already resolved to `Code` object?

Reply:
1. Hmm... You're right, we don't need this exception here. Let me yield `self.code` and return directly.
2. No. With current logic, component.code will be resolved into an arm id directly. `Code` is just a temporarily used class in
```python
with component._resolve_local_code() as code:
    component.code: str = resolver(code)
```